wayfindingsample/src/main/java/mappedin/com/wayfindingsample/MainActivity.java

```diff
@@ -204,10 +219,140 @@ public void onPause() {
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         mappedIn = new MappedIn(getApplication());
+        setContentView(R.layout.activity_main);
+
+        Toolbar toolbar =  findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        final DrawerLayout drawer =  findViewById(R.id.drawer_layout);
+        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+        drawer.setDrawerListener(toggle);
+        toggle.syncState();
+
         context = this;
         self = this;
-        setContentView(R.layout.activity_main);
+
         AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
+
+        LocationGenerator amenity = new LocationGenerator() {
+            @Override
+            public Location locationGenerator(ByteBuffer data, int _index, Venue venue){
+                return new Amenity(data, _index, venue);
+            }
+        };
+        LocationGenerator tenant = new LocationGenerator() {
+            @Override
+            public Location locationGenerator(ByteBuffer data, int _index, Venue venue){
+                return new Tenant(data, _index, venue);
+            }
+        };
+        LocationGenerator elevator = new LocationGenerator() {
+            @Override
+            public Location locationGenerator(ByteBuffer data, int _index, Venue venue){
+                return new Elevator(data, _index, venue);
+            }
+        };
+        LocationGenerator escalatorStairs = new LocationGenerator() {
+            @Override
+            public Location locationGenerator(ByteBuffer data, int _index, Venue venue){
+                return new EscalatorStairs(data, _index, venue);
+            }
+        };
+        //Only use for keys that have these location types in the binary builder
+        final LocationGenerator[] locationGenerators1 = {tenant, amenity, elevator, escalatorStairs};
+
+        LocationGenerator customerLocation = new LocationGenerator() {
+            @Override
+            public Location locationGenerator(ByteBuffer data, int _index, Venue venue){
+                return new CustomerLocation(data, _index, venue);
+            }
+        };
+        final LocationGenerator[] locationGenerators2 = {customerLocation};
+
+        // Map View
+        mapView = (MapView) getSupportFragmentManager().findFragmentById(R.id.map_fragment);
+        mapView.setDelegate(this);
+        mapComparator = new Comparator<Map>() {
```

Comment:
Any reason we moved this out of the callback?

Reply:
I was trying to improve memory usage by reducing the amount of new objects in functions we called multiple times.